moralization/transformers_model_manager.py

```diff
@@ -0,0 +1,322 @@
+    ) -> tokenization_utils_base.BatchEncoding:
+        """Creates the batches from the tokenized datasets using the Data Collator."""
+        batch = self.data_collator([item for item in tokenized_datasets["train"]])
+        return batch
+
+    def load_evaluation_metric(
+        self, label_names: List = None, eval_metric: str = "seqeval"
+    ) -> None:
+        """Loads the evaluation metric to determine scores for the training.
+
+        Args:
+            label_names (list, optional): Label names as strings for the labels to be trained. If
+            the evaluation metric is sequential evaluation, strings are required. For other metrics,
+            please consult the evaluate documentation: https://huggingface.co/docs/evaluate/choosing_a_metric
+            eval_metric (str, optional): Evaluation metric to be used, defaults to `seqeval`.
+        """
+        # default is sequential evaluation
+        # for other metrics, please see
+        # https://huggingface.co/docs/evaluate/choosing_a_metric
+        if not label_names:
+            self.label_names = ["0", "M", "M-BEG"]
+        else:
+            self.label_names = label_names
+        self.metric = evaluate.load(eval_metric)
+
+    def compute_metrics(self, eval_preds: tuple) -> dict:
+        """Convenience function to compute and return the metrics.
+
+        Args:
+            eval_preds (tuple, required): The predicted and actual labels as a tuple.
+        """
+        logits, labels = eval_preds
+        predictions = np.argmax(logits, axis=-1)
+        print(predictions)
+        # Remove ignored index (special tokens) and convert to labels
+        # we need this since seqeval operates on strings and not integers
+        true_labels = [
+            [self.label_names[m] for m in label if m != -100] for label in labels
+        ]
+        true_predictions = [
+            [self.label_names[p] for (p, m) in zip(prediction, label) if m != -100]
+            for prediction, label in zip(predictions, labels)
+        ]
+        all_metrics = self.metric.compute(
+            predictions=true_predictions, references=true_labels
+        )
+        return {
+            "precision": all_metrics["overall_precision"],
+            "recall": all_metrics["overall_recall"],
+            "f1": all_metrics["overall_f1"],
+            "accuracy": all_metrics["overall_accuracy"],
+        }
+
+    def set_id2label(self) -> None:
+        """Creates a map from label id (integer) to label name (string)."""
+
+        if not hasattr(self, "label_names"):
+            raise ValueError("Please set the label names first!")
+        self.id2label = {i: label for i, label in enumerate(self.label_names)}
+
+    def set_label2id(self) -> None:
+        """Creates a map from label name (string) to label id (integer)."""
+
+        if not hasattr(self, "id2label"):
+            raise ValueError("Please set id2label first!")
+        self.label2id = {v: k for k, v in self.id2label.items()}
```

Comment:
would it make sense to combine these two into a private function that is automatically called whenever `label_names` is set, instead of having the user call them in a specific order?

Reply:
We need this in loading the model, it needs these two functions, they are not actually accessed by the user. I put in the error message anyways, if someone would want to use them; but I don't actually see why someone would.